networkconfig/config.go

```diff
@@ -75,3 +74,25 @@ func (n NetworkConfig) SlotsPerEpoch() uint64 {
 func (n NetworkConfig) GetGenesisTime() time.Time {
 	return time.Unix(int64(n.Beacon.MinGenesisTime()), 0)
 }
+
+// Domain returns current domain type of the network
+func (n NetworkConfig) Domain() spectypes.DomainType {
+	fork := n.Beacon.ForkVersion()
+	switch fork {
+	case [4]byte{0, 0, 0, 0}:
+		if n.AlanForked(n.Beacon.EstimatedCurrentSlot()) {
+			return spectypes.AlanMainnet
+		}
+		return spectypes.GenesisMainnet
+	case [4]byte{0x01, 0x01, 0x70, 0x00}:
+		if n.AlanForked(n.Beacon.EstimatedCurrentSlot()) {
+			return spectypes.JatoAlanTestnet
+		}
+		return spectypes.JatoTestnet
+	default:
```

Comment:
@MatusKysel @moshe-blox  is this what you described? 
Im not sure herewhich beacon fork related to JatoTestnet/JatoAlanTestnet
Also should we default to mainnet or better return error?

Reply:
Another question: we need to switch a domain type at ENR after we reach the Alan epoch. I can use `func DecorateWithDomainType(domainType spectypes.DomainType)` to update it, but do we have a worker which can trigger it?

Reply:
@pavelkrolevets DomainType changes according to SSV forks, not Beacon forks

So you could use the domain type value passed to NetworkConfig on creation (such as in holesky-stage.go) and only edit the 4th byte it represents the 'fork version'

For example, if we're before AlanForkEpoch, then the version is 0, if we're after then its 1

Reply:
@pavelkrolevets 

Let's create a 'ForkListener' struct which uses a SlotTicker in the same way its used in `duties/attester.go` for example, and make it wait for the Alan fork epoch so that it could trigger the ENR update such as is done here: https://github.com/ssvlabs/ssv/blob/e865d6915005aab1d1a241bd2635ce8dfeddb1af/network/discovery/dv5_service.go#L271

Reply:
@moshe-blox great, thank you for clarification, ill update approach. Now everything clear.